sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/RowConverter.scala

```diff
@@ -3,246 +3,220 @@
 package com.azure.cosmos.spark
 
 import java.sql.{Date, Timestamp}
-import java.util
-
-import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode}
+import com.fasterxml.jackson.databind.node.{ArrayNode, BinaryNode, ObjectNode, NullNode}
 import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
-import com.microsoft.azure.cosmosdb.spark.schema.JsonSupport
 import org.apache.spark.sql.Row
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.encoders.RowEncoder
 import org.apache.spark.sql.catalyst.expressions.{GenericRowWithSchema, UnsafeMapData}
-import org.apache.spark.sql.catalyst.util.ArrayData
 
-import scala.collection.immutable.HashMap
 // scalastyle:off underscore.import
 import org.apache.spark.sql.types._
 import scala.collection.JavaConverters._
 // scalastyle:on underscore.import
 
 import org.apache.spark.unsafe.types.UTF8String
 
-// TODO: moderakh more discussion is required to decide how to do row conversion
-// see https://github.com/Azure/azure-sdk-for-java/pull/17532#discussion_r522749612 for more info
-
-// TODO add more unit tests for this class to CosmosRowConverterSpec.
-
 // scalastyle:off multiple.string.literals
 // scalastyle:off null
 object CosmosRowConverter
-  extends Serializable
-  with JsonSupport
-    with CosmosLoggingTrait {
-
-  def toInternalRow(schema: StructType, objectNode: ObjectNode): InternalRow = {
-    val row = recordAsRow(documentToMap(objectNode), schema)
-
-    RowEncoder(schema).createSerializer().apply(row)
-  }
-
-  // TODO moderakh make this configurable
-  val objectMapper = new ObjectMapper();
-
-  // TODO: moderakh this method requires a rewrite
-  // this is borrowed from old OLTP spark connector
-  def getMap(objectNode: ObjectNode): java.util.HashMap[String, AnyRef] = {
-     objectMapper.convertValue(objectNode, classOf[java.util.HashMap[String, AnyRef]])
-  }
-
-  // TODO: moderakh this method requires a rewrite
-  // this is borrowed from old OLTP spark connector
-  def documentToMap(document: ObjectNode): Map[String, AnyRef] = {
-    if (document == null) {
-      new HashMap[String, AnyRef]
-    } else {
-      getMap(document).asScala.toMap
+    extends CosmosLoggingTrait {
+
+    val objectMapper = new ObjectMapper()
```

Comment:
one of the issues we ran into with older oltp spark connector was duplicate json fields which couldn't be fetched from cosmos db by spark.

handling such scenarios require configuring how ObjectMapper and conversion works.

Please add a TODO here capturing that we need to make this behaviour configure or cover similar scenarios.

Reply:
Added TODO